### PR TITLE
chore: add day-two tests for e2e-qemu

### PR DESCRIPTION
**Pull request by TimJones**
Uses Sidero Labs d2ctl to install a small sample of
common cluster services to ensure the Talos cluster
is capable.

Signed-off-by: Tim Jones <tim.jones@siderolabs.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/5160)

### CODE REVIEW

File: hack/test/day-two/config.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    chart: ingress-nginx
+    depends:
+      - metallb
+
+  rook:
+    namespace: rook-ceph
+    repo: https://charts.rook.io/release
+    chart: rook-ceph
+
+  ceph:
```
Comment (frezbo) [inline hack/test/day-two/config.yaml:26]: it makes now sense to update the `e2e-qemi-csi` tests and use the charts for install, and move the tests somewhere with day2 tests

Comment (frezbo) [reply hack/test/day-two/config.yaml:26]: https://github.com/talos-systems/talos/blob/master/hack/test/e2e.sh#L228

Comment (TimJones) [reply hack/test/day-two/config.yaml:26]: Once I've confirmed the day-two tests work and cover the same case, I will remove the `e2e-qemu-csi` tests. I just didn't want to break anyone using them in the mean time :)

**Review by frezbo — APPROVED** [review]

**Review by smira — COMMENTED** [review]
:tada:

**Pull request merged**